src/mscorlib/src/System/Security/SafeBSTRHandle.cs

```diff
@@ -13,7 +13,7 @@ internal sealed class SafeBSTRHandle : SafeBuffer
 
         internal static SafeBSTRHandle Allocate(uint lenInChars)
         {
-            uint lenInBytes = lenInChars * sizeof(char);
+            ulong lenInBytes = lenInChars * sizeof(char);
```

Comment:
This is not fixing anything. The multiplication will be performed as uint, and only then the result gets cast to ulong. So if there is an integer overflow during the multiplication, you will get the exact same result as before. Instead, you would need to add the cast to at least one of the multiplicants.